service/src/test/java/bio/terra/workspace/service/workspace/WorkspaceServiceTest.java

```diff
@@ -481,6 +515,86 @@ void createGoogleContextRawlsStageThrows() throws Exception {
                 request.getWorkspaceId(), jobId, USER_REQUEST, "/fake/value"));
   }
 
+  @Test
+  public void cloneGcpWorkspace() {
```

Comment:
Is this really a connected test? It looks like you have mocked out everything that touches the cloud platform.

Reply:
I think the problem is more general. This test started just testing workspace. Then it grew some cloud context tests. Then it grew a delete workspace test with a reference to make sure we deleted those. Now it is growing to actually create a controlled object. I am thinking this test shouldn't be in this module.

We would have a module that has only basic workspace tests, like duplicate name and bad name, etc. Keep that separate from tests that create resources.

Reply:
That's a good question in general about the utility of connected vs integration tests. I don't really know the strategy (and certainly didn't when I initially filed the ticket). Cloning is slow, so this adds considerably to the test burden.

We could split out more complex tests from this test class (if that's what you mean by module), but that would introduce some duplication in the form of the application context setup. Maybe we can make a shareable app context for this kind of test, but we're getting out of scope for this PR. I did add a bucket clone only test to the `ControlledResourceServiceTest`, but I thought cloning a whole workspace wouldn't fit there (as it might bring in other dependencies). So it's a square peg either way.

Reply:
Agreed that this is a square peg and agree that we do not have a clean separation of connected and integration tests. Let’s go ahead with this one as is. I think it would be worth sorting out the tests and maybe running different sets as PR vs nightly tests.